source/elements/oneMKL/source/domains/blas/syrk_batch.rst

```diff
@@ -0,0 +1,528 @@
+
+The group API operation is defined as:
+::
+
+   idx = 0
+   for i = 0 … group_count – 1
+       for j = 0 … group_size – 1
+           A, B, and C are matrices in a[idx] and c[idx]
+           C := alpha[i] * op(A) * op(A)^T + beta[i] * C
+           idx = idx + 1
+       end for
+   end for
+
+The strided API operation is defined as
+::
+
+   for i = 0 … batch_size – 1
+       A, B and C are matrices at offset i * stridea, i * stridec in a and c.
+       C := alpha * op(A) * op(A)^T + beta * C
+   end for
+
+where:
+
+op(X) is one of op(X) = X, or op(X) = X\ :sup:`T`, or op(X) = X\ :sup:`H`,
+
+``alpha`` and ``beta`` are scalars,
+
+``A`` and ``C`` are matrices,
+
+op(``A``) is ``n`` x ``k`` and ``C`` is ``n`` x ``n``.
+
+ 
+For group API, ``a`` and ``c`` arrays contain the pointers for all the input matrices. 
+The total number of matrices in ``a`` and ``c`` are given by: 
+
+.. math::
+
+      total\_batch\_count = \sum_{i=0}^{group\_count-1}group\_size[i]    
+ 
+For strided API, ``a`` and ``c`` arrays contain all the input matrices. The total number of matrices 
+in ``a`` and ``c`` are given by the ``batch_size`` parameter.  
+   
+**Group API**
+
+.. rubric:: Syntax
+   
+.. code-block:: cpp
+
+   namespace oneapi::mkl::blas::column_major {
+       sycl::event syrk_batch(sycl::queue &queue,
+                              uplo *upper_lower,
+                              transpose *trans,
+                              std::int64_t *n,
+                              std::int64_t *k,
+                              T *alpha,
+                              const T **a,
+                              std::int64_t *lda,
+                              T *beta,
+                              T **c,
+                              std::int64_t *ldc,
+                              std::int64_t group_count,
+                              std::int64_t *group_size,
+                              const std::vector<sycl::event> &dependencies = {})
+   }
+.. code-block:: cpp
+
+   namespace oneapi::mkl::blas::row_major {
+       sycl::event syrk_batch(sycl::queue &queue,
+                              uplo *upper_lower,
+                              transpose *trans,
+                              std::int64_t *n,
+                              std::int64_t *k,
+                              T *alpha,
+                              const T **a,
+                              std::int64_t *lda,
+                              T *beta,
+                              T **c,
+                              std::int64_t *ldc,
+                              std::int64_t group_count,
+                              std::int64_t *group_size,
+                              const std::vector<sycl::event> &dependencies = {})
+   }
+
+.. container:: section
+
+   .. rubric:: Input Parameters
+
+   queue
+      The queue where the routine should be executed.
+
+   upper_lower
+      Spcifies whether an array of data in ``C`` is stored in its upper or lower triangle.
```

Comment:
Array of ``group_count`` ``onemkl::upper_lower`` values. ``upper_lower[i]'' specifies whether data in ``C`` for every matrix in group ``i`` is in its upper or lower triangle.

Reply:
adjusted